shared/request_caching.go

```diff
@@ -75,8 +77,13 @@ type cachingHandler struct {
 	cache       ReadWritable
 	isCacheable func(*http.Request) bool
 	getCacheKey func(*http.Request) interface{}
+	shouldCache func(int) bool
 }
 
+func defaultIsCacheable(*http.Request) bool          { return true }
+func defaultGetCacheKey(r *http.Request) interface{} { return r.URL.String() }
+func defaultShouldCache(status int) bool             { return status == http.StatusOK }
```

Comment:
Guess I didn't have status-based/output 'cachability' before, but same pattern applies. If you like the idea that the name implies the behaviour (instead of saying "I'm the default" and reader must work it out an extra reading step away), then consolidate whatever names you think are good - I didn't scrutinize my naming, just the implications of the naming.